optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/domain/tenant/view/RosterConstraintConfigurationView.java

```diff
@@ -0,0 +1,244 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.domain.tenant.view;
+
+import java.time.DayOfWeek;
+
+import javax.validation.constraints.NotNull;
+
+import org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScore;
+import org.optaweb.employeerostering.domain.common.AbstractPersistable;
+
+public class RosterConstraintConfigurationView extends AbstractPersistable {
+
+    @NotNull
+    private Integer undesiredTimeSlotWeight = 100;
+    @NotNull
+    private Integer desiredTimeSlotWeight = 10;
+    @NotNull
+    private Integer rotationEmployeeMatchWeight = 500;
+    @NotNull
+    private DayOfWeek weekStartDay = DayOfWeek.MONDAY;
+
+    // COVID-specific constraints
+    private HardMediumSoftLongScore lowRiskEmployeeInCovidWardMatchWeight = HardMediumSoftLongScore.ofSoft(1);
+    private HardMediumSoftLongScore moderateRiskEmployeeInCovidWardMatchWeight = HardMediumSoftLongScore.ofSoft(5);
+    private HardMediumSoftLongScore highRiskEmployeeInCovidWardMatchWeight = HardMediumSoftLongScore.ofSoft(10);
+    private HardMediumSoftLongScore extremeRiskEmployeeInCovidWardMatchWeight = HardMediumSoftLongScore.ofHard(1);
+
+    private HardMediumSoftLongScore requiredSkill = HardMediumSoftLongScore.ofHard(100);
+    private HardMediumSoftLongScore unavailableTimeSlot = HardMediumSoftLongScore.ofHard(50);
+    private HardMediumSoftLongScore oneShiftPerDay = HardMediumSoftLongScore.ofHard(10);
+    private HardMediumSoftLongScore noShiftsWithinTenHours = HardMediumSoftLongScore.ofHard(1);
+    private HardMediumSoftLongScore contractMaximumDailyMinutes = HardMediumSoftLongScore.ofHard(1);
+    private HardMediumSoftLongScore contractMaximumWeeklyMinutes = HardMediumSoftLongScore.ofHard(1);
+    private HardMediumSoftLongScore contractMaximumMonthlyMinutes = HardMediumSoftLongScore.ofHard(1);
+    private HardMediumSoftLongScore contractMaximumYearlyMinutes = HardMediumSoftLongScore.ofHard(1);
+
+    private HardMediumSoftLongScore assignEveryShift = HardMediumSoftLongScore.ofMedium(1);
+
+    private HardMediumSoftLongScore undesiredTimeSlot = HardMediumSoftLongScore.ofSoft(1);
+    private HardMediumSoftLongScore desiredTimeSlot = HardMediumSoftLongScore.ofSoft(1);
+    private HardMediumSoftLongScore notRotationEmployee = HardMediumSoftLongScore.ofSoft(1);
+
+    @SuppressWarnings("unused")
+    public RosterConstraintConfigurationView() {
+        super(-1);
+    }
+
+    public RosterConstraintConfigurationView(Integer tenantId,
```

Comment:
No Covid weights in this constructor? Is this constructor used?

Reply:
See above.